### ISSUE TITLE
Docker runtime would overwirte memory limit of its cgroup parent when it starts to run

### ISSUE THREAD

**Issue by zq-david-wang** [moby/moby#33876]
**Description**
I am trying to reserve system resource to make sure the no docker application would eat up all system resource. My system is running docker 1.12.6 on Centos7.3, systemd cgroup drive.
In docker service unit file, I create a system-docker.slice
> ExecStartPre=/usr/local/bin/docker-cgroup-setup.sh

The setup script would do something like this:
> echo 1-7 > /sys/fs/cgroup/cpuset/system.slice/system-docker.slice/cpuset.cpus
echo 31343419392 > /sys/fs/cgroup/memory/system.slice/system-docker.slice/memory.limit_in_bytes

I meant to reserve cpu-0 and some extra memory(2G) for the system in case docker applications run into some problem.

And relevant  docker service options to make use this cgroup slice:
> --cgroup-parent=system-docker.slice --exec-opt native.cgroupdriver=systemd

After power cycle the system, before running any docker application, the content in /sys/fs/cgroup/memory/system.slice/system-docker.slice/memory.limit_in_bytes is what I expected.
But when I run a container via "docker run", the system-docker.slice for memory would be overwritten to a big value(9223372036854771712) indicating unlimited memory access.

**Steps to reproduce the issue:**
1.  mkdir -p /sys/fs/cgroup/memory/system.slice/system-docker.slice
2.  echo 1073741824 > /sys/fs/cgroup/memory/system.slice/system-docker.slice/memory.limit_in_bytes
3.  configure docker service unit file to use option: --cgroup-parent=system-docker.slice --exec-opt native.cgroupdriver=systemd
4.  run 'systemctl start docker' (Reboot the system if docker runtime has already been started at least once)
5.  start any docker container via "docker run"

**Describe the results you received:**
/sys/fs/cgroup/memory/system.slice/system-docker.slice/memory.limit_in_bytes is over written with maximum value

**Describe the results you expected:**
The parent cgroup should keep its original limit settings.

**Additional information you deem important (e.g. issue happens only occasionally):**
I use oci hook as a workaround to reset the limit before starting each container, the solution is not clean to me. 
**Output of `docker version`:**

```
Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-common-1.12.6-11.el7.centos.x86_64
 Go version:      go1.7.4
 Git commit:      96d83a5/1.12.6
 Built:           Tue Mar  7 09:23:34 2017
 OS/Arch:         linux/amd64

Server:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-common-1.12.6-11.el7.centos.x86_64
 Go version:      go1.7.4
 Git commit:      96d83a5/1.12.6
 Built:           Tue Mar  7 09:23:34 2017
 OS/Arch:         linux/amd64

```

**Output of `docker info`:**
ontainers: 32
 Running: 24
 Paused: 0
 Stopped: 8
Images: 191
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-253:0-268447019-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 55.1 GB
 Data Space Total: 107.4 GB
 Data Space Available: 52.28 GB
 Metadata Space Used: 73.43 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.074 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /data/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /data/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: bridge host overlay null
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Security Options: seccomp
Kernel Version: 3.10.0-514.16.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 8
Total Memory: 31.19 GiB
Name: ennew-154
ID: XFYM:ZH77:BDOV:AVYS:GQQS:6HSY:7S6I:77DH:OCEB:73HZ:Z56Z:V3E5
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 10.19.132.184:30100
 127.0.0.0/8
Registries: docker.io (secure)

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
It looks like you're running the Red Hat fork of docker, not the official packages; also, docker 1.12 is outdated and no longer maintained.

The systemd cgroups are known to be buggy, which is the reason the official packages also default to using the `cgroupfs` driver.

I'm closing this because the Red Hat fork of docker is not maintained here, but let us know if you're having the same problem using an updated version of the official packages
